integration/chunks_delete_series_test.go

```diff
@@ -0,0 +1,221 @@
+			require.NoError(t, err)
+			require.Equal(t, 200, res.StatusCode)
+
+			seriesToPush[i].pushedVectors = append(seriesToPush[i].pushedVectors, expectedVector)
+		}
+
+		// lets make ingester flush the chunks immediately to the store.
+		res, err := e2e.GetRequest("http://" + ingester.HTTPEndpoint() + "/flush")
+		require.NoError(t, err)
+		require.Equal(t, 204, res.StatusCode)
+
+		// Let's wait until all chunks are flushed.
+		require.NoError(t, ingester.WaitSumMetrics(e2e.Equals(0), "cortex_ingester_flush_queue_length"))
+		require.NoError(t, ingester.WaitSumMetrics(e2e.Equals(0), "cortex_ingester_flush_series_in_progress"))
+	}
+
+	// call flush again because chunks are sometimes still retained in memory due to chunks flush operation being async while chunk cleanup from memory is not.
+	res, err := e2e.GetRequest("http://" + ingester.HTTPEndpoint() + "/flush")
+	require.NoError(t, err)
+	require.Equal(t, 204, res.StatusCode)
+
+	require.NoError(t, ingester.WaitSumMetrics(e2e.Equals(0), "cortex_ingester_memory_chunks"))
+
+	// start a purger
+	purger := e2ecortex.NewPurger("purger", flags, "")
+	purger.HTTPService.SetEnvVars(bigtableFlag)
+
+	require.NoError(t, s.StartAndWaitReady(purger))
+
+	// perform deletion for 2 days interval for each week covering all the Index Stores.
+	deletionDuration := 2 * 24 * time.Hour
+	deletedIntervals := intervals{}
+	endpoint := "http://" + purger.HTTPEndpoint() + "/prometheus/api/v1/admin/tsdb/delete_series?match[]=delete_series{common=\"label\"}&start=%f&end=%f"
+	for ts := now; ts.After(oldestStoreStartTime); ts = ts.Add(-perStoreDuration) {
+		// expand the interval by a second on both the ends since the requests are getting aligned with pushed samples at ms precision which
+		// sometimes does not match exactly in purger during parsing time from a string.
+		deletedInterval := interval{e2e.TimeToMilliseconds(ts.Add(-(deletionDuration + time.Second))), e2e.TimeToMilliseconds(ts.Add(time.Second))}
+		deletedIntervals = append(deletedIntervals, deletedInterval)
+
+		res, err := client.PostRequest(fmt.Sprintf(endpoint, float64(deletedInterval.start)/float64(1000), float64(deletedInterval.end)/float64(1000)), nil)
+		require.NoError(t, err)
+		require.Equal(t, 204, res.StatusCode)
+	}
+
+	// check whether purger has received expected number of delete requests.
+	require.NoError(t, purger.WaitSumMetrics(e2e.Equals(float64(len(deletedIntervals))), "cortex_purger_delete_requests_received_total"))
+
+	// stop the purger and recreate it since we load requests for deletion every hour.
+	require.NoError(t, s.Stop(purger))
+
+	purger = e2ecortex.NewPurger("purger", mergeFlags(flags, map[string]string{
+		"-purger.delete-request-cancel-period": "-1m", // we retain delete requests for a day(default)+1m, changing it to a negative value so that delete requests get picked up immediately.
+	}), "")
+	purger.HTTPService.SetEnvVars(bigtableFlag)
+	require.NoError(t, s.StartAndWaitReady(purger))
+
+	time.Sleep(time.Second)
```

Comment:
I was testing this PR and this makes the test flaky (because may need more than 1 second). Let me get https://github.com/cortexproject/cortex/pull/2522 merged and then I will commit a change to this PR to fix it based on the PR #2522 I'm currently working on.

Reply:
Fixed it!